shell/platform/embedder/embedder_asset_resolver.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/embedder/embedder_asset_resolver.h"
+#include "flutter/shell/platform/embedder/embedder_struct_macros.h"
+
+namespace flutter {
+
+class EmbedderAssetResolver final : public flutter::AssetResolver {
+ public:
+  explicit EmbedderAssetResolver(void* user_data,
```

Comment:
The C style API must be limited to `embedder.h` and `embedder.cc`. Prefer an `std::function` here. The baton can be captured. Then, you don't have to have separate ivars for the baton and the callback. The engine doesn't care about the baton ever anyway.

Reply:
Better yet, make the caller return a valid mapping.

So something like: `using Callback = std::function<std::unique_ptr<fml::Mapping>(const std::string& asset_name)>;`